src/overlays/actors/ovl_Obj_Lightswitch/z_obj_lightswitch.c

```diff
@@ -109,89 +104,58 @@ void ObjLightswitch_UpdateSwitchFlags(ObjLightswitch* this, GlobalContext* globa
+        tempResultDiff = tempResult - 30.0f;
 
         if (tempResult > 30.0f) {
             tempResult = 30.0f;
         } else {
-            tempResult = 900.0f - SQ(tempResult_shorter);
-            if (tempResult < 100.0f) {
-                tempResult = 100.0f;
-            }
+            tempResult = SQ(30.0f) - SQ(tempResultDiff);
+            tempResult = CLAMP_MIN(tempResult, SQ(10.0f));
             tempResult = sqrtf(tempResult);
         }
 
-        tempResult5 = 2.0f * ((Rand_ZeroOne() - 0.5f) * tempResult);
-
-        // permuter seems to think something could be added to fix ordering here, but doesn't fix enough
-        // goto dummy_label_809848; dummy_label_809848: ; // minor regalloc fix
-        // if (cosResult){} // minor regalloc fix, but breaks some code
-
-        // saving fabsf result as a temp might help, but need to reduce temps
-        // fabsResult = fabsf(tempResult5);
-        // randomOutput = (Rand_ZeroOne() * 10.0f) + ((30.0f - fabsResult) * 0.5f);
-        // randomOutput = fabsf(tempResult5);
-        // randomOutput = (Rand_ZeroOne() * 10.0f) + ((30.0f - randomOutput) * 0.5f);
-        randomOutput = (Rand_ZeroOne() * 10.0f) + ((30.0f - fabsf(tempResult5)) * 0.5f);
-
-        effectPos.x = this->actor.world.pos.x + ((randomOutput * sinResult) + (tempResult5 * cosResult));
-        effectPos.y = this->actor.world.pos.y + tempResult_shorter + 10.0f;
-        effectPos.z = this->actor.world.pos.z + ((randomOutput * cosResult) - (tempResult5 * sinResult));
-        // weirdly, using a temp here make a big difference, but.. is that right? just one temp? prob ordering issue
-        // above
-        // tempResult = this->actor.world.pos.z + ((randomOutput * cosResult) - (tempResult5 * sinResult));
-        // effectPos.z = tempResult;
-
-        EffectSsDeadDb_Spawn(globalCtx, &effectPos, &D_801D15B0, &D_801D15B0, &lightswitchEffectPrimColor,
-                             &lightswitchEffectEnvColor, 100, 0, 9);
+        tempResult = 2.0f * ((Rand_ZeroOne() - 0.5f) * tempResult);
+        rand = ((30.0f - fabsf(tempResult)) * 0.5f) + (Rand_ZeroOne() * 10.0f);
+
+        effectPos.x = this->actor.world.pos.x + ((rand * sinResult) + (tempResult * cosResult));
+        effectPos.y = this->actor.world.pos.y + tempResultDiff + 10.0f;
+        effectPos.z = this->actor.world.pos.z + ((rand * cosResult) - (tempResult * sinResult));
+
+        EffectSsDeadDb_Spawn(globalCtx, &effectPos, &D_801D15B0, &D_801D15B0, &sLightswitchEffectPrimColor,
+                             &sLightswitchEffectEnvColor, 100, 0, 9);
     }
 }
-#else
-void ObjLightswitch_SpawnEffects(ObjLightswitch* this, GlobalContext* globalCtx);
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Lightswitch_0x8095FBF0/ObjLightswitch_SpawnEffects.asm")
-#endif
 
 void ObjLightswitch_Init(Actor* thisx, GlobalContext* globalCtx) {
     ObjLightswitch* this = THIS;
     s32 pad;
-    u32 switchFlagResult;
+    u32 isGetSwitchSuccess;
```

Comment:
I wasn't strongly attached to `switchFlagResult` but... is this really a better temp name?

Maybe `isSwitchTriggered` or `isSwitchActivated` ?

Reply:
Not my best name. I'll go with `isSwitchActivated`